### PR TITLE
Linked list tests

### PR DESCRIPTION
**Pull request by Parowicz**
# Welcome to Dart community

I've changed github action to run all tests from all folders */*.dart and subfolders */*/*.dart. I don't have ideas how to force dart to traverse whole directory tree.

Also I've encountered some mirror bugs. Most of them is fixed, but I'll have to spend more time on remaining ones.

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (cclauss) [inline pubspec.yaml:1]:
```suggestion
```

**Comment by cclauss** [comment]
AWESOME!!

**Comment by cclauss** [comment]
Should we put [`@Skip('currently failing (see issue #86)')`](https://pub.dev/packages/test#skipping-tests) at the top of `math/fermats_little_theorem.dart`?

**Comment by Parowicz** [comment]
Unfortunate it's compilation error. I can fix types in this file and raise generic fail until issue is resolved.

**Comment by cclauss** [comment]
So the @Skip does not prevent `dart test` from compiling the file?

**Comment by Parowicz** [comment]
Nvm it's just too late. It will work as you say.

**Comment by Parowicz** [comment]
Ok, now it looks fine. Using *.yaml config files solved all the problems.

**Comment by cclauss** [comment]
VERY IMPRESSIVE!!

**Comment by cclauss** [comment]
Ready for Review?

**Review by cclauss — APPROVED** [review]
LGTM!!!

**Pull request merged**